### PR TITLE
[Improve]Use Jackson replace Fastjson

### PR DESCRIPTION
**Pull request by CalvinKirs**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-feishu/src/main/scala/org/apache/seatunnel/spark/feishu/FeishuClient.scala [hunk 1]
```diff
@@ -16,25 +16,27 @@
...
 import org.slf4j.{Logger, LoggerFactory}
 
 class FeishuClient(appId: String, appSecret: String) {
-  val logger: Logger = LoggerFactory.getLogger(this.getClass)
+  /*val logger: Logger = LoggerFactory.getLogger(this.getClass)
 
   def getToken: String = {
     val url = Config.TOKEN_URL.format(appId, appSecret)
     val result = this.requestFeishuApi(url, null)
     logger.info(s"Request token and get result $result")
```
Comment (CalvinKirs) [inline seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-feishu/src/main/scala/org/apache/seatunnel/spark/feishu/FeishuClient.scala]: I don't familiarize Scala, anyone could update this

File: seatunnel-apis/seatunnel-api-flink/pom.xml [hunk 2]
```diff
@@ -87,6 +93,11 @@
             <artifactId>flink-avro</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+        </dependency>
```
Comment (ruanwenjun) [inline seatunnel-apis/seatunnel-api-flink/pom.xml]: We have defined the jackson-databind in `seatunnel-common`, so we don't need to define in this pom again.

File: seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/stream/FlinkStreamExecution.java [hunk 3]
```diff
@@ -99,7 +99,7 @@ private Optional<DataStream<Row>> fromSourceTable(Config pluginConfig) {
         if (pluginConfig.hasPath(SOURCE_TABLE_NAME)) {
             StreamTableEnvironment tableEnvironment = flinkEnvironment.getStreamTableEnvironment();
             Table table = tableEnvironment.scan(pluginConfig.getString(SOURCE_TABLE_NAME));
-            return Optional.ofNullable(TableUtil.tableToDataStream(tableEnvironment, table, false));
+            return Optional.ofNullable(TableUtil.tableToDataStream(tableEnvironment, table, true));
```
Comment (ruanwenjun) [inline seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/stream/FlinkStreamExecution.java]: This file need to revert?

File: seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/SchemaUtil.java [hunk 4]
```diff
@@ -123,12 +124,12 @@ private static void getJsonSchema(Schema schema, JSONObject json) {
...
                 schema.field(key, Types.JAVA_BIG_DEC());
-            } else if (value instanceof JSONObject) {
-                schema.field(key, getTypeInformation((JSONObject) value));
-            } else if (value instanceof JSONArray) {
-                Object obj = ((JSONArray) value).get(0);
-                if (obj instanceof JSONObject) {
-                    schema.field(key, ObjectArrayTypeInfo.getInfoFor(Row[].class, getTypeInformation((JSONObject) obj)));
+            } else if (value instanceof ObjectNode) {
+                schema.field(key, getTypeInformation((ObjectNode) value));
+            } else if (value instanceof ObjectNode) {
```
Comment (ruanwenjun) [inline seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/SchemaUtil.java]:
```suggestion
            } else if (value instanceof ObjectNode) {
                schema.field(key, getTypeInformation((ObjectNode) value));
            } else if (value instanceof ArrayNode) {
```

File: seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/TableUtil.java [hunk 5]
```diff
@@ -39,7 +39,11 @@ public static DataStream<Row> tableToDataStream(StreamTableEnvironment tableEnvi
         if (isAppend) {
             return tableEnvironment.toAppendStream(table, typeInfo);
         }
-        return tableEnvironment.toChangelogStream(table);
+        return tableEnvironment
+                .toRetractStream(table, typeInfo)
+                .filter(row -> row.f0)
+                .map(row -> row.f1)
+                .returns(typeInfo);
```
Comment (ruanwenjun) [inline seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/TableUtil.java]: We need to revert this file?

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/JsonUtils.java [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+     * @param object  object
+     * @param feature feature
+     * @return object to json string
+     */
+    public static String toJsonString(Object object, SerializationFeature feature) {
+        try {
+            ObjectWriter writer = OBJECT_MAPPER.writer(feature);
+            return writer.writeValueAsString(object);
+        } catch (Exception e) {
+            LOGGER.error("object to json exception!", e);
```
Comment (ruanwenjun) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/JsonUtils.java]: It's better to keep the same logic to deal with exception. In this file, there are some methods will catch the exception and some methods will re throw the exception.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: seatunnel-apis/seatunnel-api-flink/pom.xml (hunk 2 above)
Comment (CalvinKirs) [reply seatunnel-apis/seatunnel-api-flink/pom.xml]: done~

File: seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/TableUtil.java (hunk 5 above)
Comment (CalvinKirs) [reply seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/util/TableUtil.java]: yup,from #1972

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/JsonUtils.java (hunk 6 above)
Comment (CalvinKirs) [reply seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/JsonUtils.java]: Good catch

File: seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-feishu/src/main/scala/org/apache/seatunnel/spark/feishu/FeishuClient.scala (hunk 1 above)
Comment (ruanwenjun) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-feishu/src/main/scala/org/apache/seatunnel/spark/feishu/FeishuClient.scala]: I finished the scala code.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**